retry/retry.go

```diff
@@ -0,0 +1,15 @@
+package retry
+
+import "time"
+
+func Do(f func() error, maxRuns, sleepMs int) error {
```

Comment:
I already have a flexible implementation of this in `nmagent/internal` that is probably worth promoting to this package if you need it too. It can support constant backoff, exponental, linear, maximums retries, whatever you like. It also respects contexts and deals with errors.